File: schemaorg-rs/src/shipping_rate_settings.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///https://schema.org/additionalType
///https://schema.org/Text
///https://schema.org/URL
pub type ShippingRateSettingsAdditionalTypeFieldEnum = String;
///https://schema.org/ShippingRateSettings
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct ShippingRateSettings {
    #[serde(rename = "@context")]
    pub context: String,
    ///https://schema.org/shippingDestination
    #[serde(rename = "shippingDestination")]
    #[serde_as(as = "OneOrMany<_>")]
    pub shipping_destination: Vec<DefinedRegion>,
    ///https://schema.org/url
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///https://schema.org/description
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<ShippingRateSettingsDescriptionFieldEnum>,
    ///https://schema.org/disambiguatingDescription
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///https://schema.org/identifier
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<ShippingRateSettingsIdentifierFieldEnum>,
    ///https://schema.org/sameAs
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///https://schema.org/image
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<ShippingRateSettingsImageFieldEnum>,
    ///https://schema.org/additionalType
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<ShippingRateSettingsAdditionalTypeFieldEnum>,
    ///https://schema.org/isUnlabelledFallback
    #[serde(rename = "isUnlabelledFallback")]
    #[serde_as(as = "OneOrMany<_>")]
    pub is_unlabelled_fallback: Vec<String>,
    ///https://schema.org/weightPercentage
    #[serde(rename = "weightPercentage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub weight_percentage: Vec<f32>,
    ///https://schema.org/orderPercentage
    #[serde(rename = "orderPercentage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub order_percentage: Vec<f32>,
    ///https://schema.org/mainEntityOfPage
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<ShippingRateSettingsMainEntityOfPageFieldEnum>,
    ///https://schema.org/doesNotShip
    #[serde(rename = "doesNotShip")]
    #[serde_as(as = "OneOrMany<_>")]
    pub does_not_ship: Vec<String>,
    ///https://schema.org/shippingRate
    #[serde(rename = "shippingRate")]
    #[serde_as(as = "OneOrMany<_>")]
    pub shipping_rate: Vec<ShippingRateSettingsShippingRateFieldEnum>,
    ///https://schema.org/freeShippingThreshold
    #[serde(rename = "freeShippingThreshold")]
    #[serde_as(as = "OneOrMany<_>")]
    pub free_shipping_threshold: Vec<ShippingRateSettingsFreeShippingThresholdFieldEnum>,
    ///https://schema.org/alternateName
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///https://schema.org/name
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///https://schema.org/subjectOf
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<ShippingRateSettingsSubjectOfFieldEnum>,
    ///https://schema.org/potentialAction
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
}

File: schemaorg-ts/src/shipping_rate_settings.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type ShippingRateSettingsAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/ShippingRateSettings>
 **/
export interface ShippingRateSettings {
  "@context": string;
  "@type": "ShippingRateSettings";

  /**
   * <https://schema.org/shippingRate>
   **/
  shippingRate: all.OneOrMany<all.ShippingRateSettingsShippingRateFieldEnum>;

  /**
   * <https://schema.org/orderPercentage>
   **/
  orderPercentage: all.OneOrMany<number>;

  /**
   * <https://schema.org/freeShippingThreshold>
   **/
  freeShippingThreshold: all.OneOrMany<all.ShippingRateSettingsFreeShippingThresholdFieldEnum>;

  /**
   * <https://schema.org/weightPercentage>
   **/
  weightPercentage: all.OneOrMany<number>;

  /**
   * <https://schema.org/doesNotShip>
   **/
  doesNotShip: all.OneOrMany<string>;

  /**
   * <https://schema.org/shippingDestination>
   **/
  shippingDestination: all.OneOrMany<all.DefinedRegion>;

  /**
   * <https://schema.org/isUnlabelledFallback>
   **/
  isUnlabelledFallback: all.OneOrMany<string>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.ShippingRateSettingsMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.ShippingRateSettingsIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.ShippingRateSettingsDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.ShippingRateSettingsSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.ShippingRateSettingsAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.ShippingRateSettingsImageFieldEnum>;
}
